pkg/networkserver/grpc_gsns.go

```diff
@@ -1276,6 +1276,37 @@ func (ns *NetworkServer) HandleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 
 // ReportTxAcknowledgment is called by the Gateway Server when a tx acknowledgment arrives.
 func (ns *NetworkServer) ReportTxAcknowledgment(ctx context.Context, up *ttnpb.GatewayTxAcknowledgment) (_ *pbtypes.Empty, err error) {
-	// TODO: Send ApplicationDownlinkSent event from Tx acknowledgment (https://github.com/TheThingsNetwork/lorawan-stack/issues/56)
+	ack := up.GetTxAck()
+	if ack.GetResult() != ttnpb.TxAcknowledgment_SUCCESS {
+		return ttnpb.Empty, nil
+	}
+	down, err := ns.scheduledDownlinkMatcher.Match(ctx, ack)
+	if err != nil {
+		return nil, err
```

Comment:
I think we will see quite some `not_found` errors on the GS side if we report the local `not_found` error if the TTL expires. Nonetheless, I think this is not terrible.

Consider making the timeout part of the NS config - at least we can configure this in production then.